eng/common/scripts/ChangeLog-Operations.ps1

```diff
@@ -121,10 +119,117 @@ function Confirm-ChangeLogEntry {
   return $true
 }
 
-function Set-TestChangeLog($TestVersion, $changeLogFile, $ReleaseEntry) {
-  Set-Content -Path $changeLogFile -Value @"
-# Release History
-## $TestVersion ($(Get-Date -f "yyyy-MM-dd"))
-- $ReleaseEntry
-"@
+function New-ChangeLogEntry {
+  param (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [String]$Version,
+    [String]$Status="(Unreleased)",
+    [String]$Title,
+    [String[]]$Content
+  )
+
+  # Validate relase Status
+  $Status = $Status.Trim().Trim("()")
+  if ($Status -ne "Unreleased") {
+    $dateFormat = "yyyy-MM-dd"
+    $provider = [System.Globalization.CultureInfo]::InvariantCulture
+    try {
+      $Status = ([System.DateTime]::ParseExact($Status, $dateFormat, $provider)).ToString($dateFormat)
+    }
+    catch {
+        LogWarning "Invalid date [ $Status ] passed as status for Version [$Version]. Please use a valid date in the format '$dateFormat' or use '(Unreleased)'"
+    }
+  }
+
+  $Status = "($Status)"
+  if (!$Content) { $Content = @() }
+  if (!$Title) { $Title = "## $Version $Status" }
+
+  $newChangeLogEntry = [pscustomobject]@{ 
+    ReleaseVersion = $Version
+    ReleaseStatus  = $Status
+    ReleaseTitle   = $Title
+    ReleaseContent = $Content
+  }
+
+  return $newChangeLogEntry
+}
+
+function Add-ChangeLogEntry {
```

Comment:
What value does this function really have? It seems like it would be pretty easy to add to a hash that you already have.